src/api/cached_client.rs

```diff
@@ -324,7 +324,17 @@ impl SpotifyApiClient for CachedSpotifyClient {
             let mut album: AlbumDescription = album?.into();
             album.is_liked = liked?[0];
 
-            Ok(album)
+            let info = self
+                .cache_get_or_write(SpotCacheKey::Album(&id), None, |etag| {
+                    self.client.get_album_info(&id).etag(etag).send()
```

Comment:
Almost there :)

See the issue here is that we end up calling two times the same album endpoint within this function -- it's alright because of the cache but it should not happen regardless (actually not fine, it writes two different models to the same cache entry but anyway)

Ask yourself, what does the v1/album endpoint actually return? a "full" album as defined by Spotify, so the union of your current AlbumInfo + the previous Album model

Just update the get_album fn from the client to return the full API model, and then you're good I think, this line is no longer needed :) nor is get_album_info needed